src/main.rs

```diff
@@ -342,3 +307,75 @@ fn err_to_warp(e: podsync::Error) -> impl warp::Reply {
 pub struct QuerySince {
     since: crate::time::Timestamp,
 }
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+enum UsernameFormat {
+    Name,
+    NameJson,
+}
+impl UsernameFormat {
+    pub fn convert<'a>(&self, username: &'a str) -> Result<&'a str, warp::Rejection> {
+        match self {
+            Self::Name => Ok(username),
+            Self::NameJson => split_format_json(username).map_err(warp::reject::custom),
+        }
+    }
+}
+
+fn cookie_authorize(
+    username_fmt: UsernameFormat,
+    podsync: Arc<PodSync>,
+) -> impl Filter<Extract = (PodSyncAuthed<true>,), Error = warp::Rejection> + Clone {
+    let cookie_auth_check = warp::cookie(COOKIE_NAME).and_then({
+        move |session_id: SessionId| {
+            let podsync = Arc::clone(&podsync);
+
+            async move {
+                podsync
+                    .authenticate(session_id)
+                    .await
+                    .map_err(warp::reject::custom)
+            }
+        }
+    });
+
+    warp::path::param::<String>()
+        .and(cookie_auth_check)
+        .and_then(move |username: String, podsync: PodSyncAuthed| async move {
+            podsync
+                .with_user(username_fmt.convert(&username)?)
+                .map_err(warp::reject::custom)
+        })
+}
+
+fn login_authorize(
+    username_fmt: UsernameFormat,
+    podsync: Arc<PodSync>,
+) -> impl Filter<Extract = (PodSyncAuthed<true>,), Error = warp::Rejection> + Clone {
+    warp::path::param::<String>()
+        .and(warp::header("authorization"))
+        .and(warp::cookie::optional(COOKIE_NAME))
```

Comment:
Here we can leave out the cookie - a `session_id` cookie along with an auth header is just for the `/login` endpoint (so clients can [verify if their session id is still valid](https://github.com/gpodder/mygpo/blob/HEAD/doc/api/reference/auth.rst?plain=1#L19-L20)), and otherwise we're only interested in the authorization header (so we can pass `None` to `.login()`)

Reply:
Check; That was what I had first, but I figured It couldn't really hurt, but you are correct.